stream-projects/chat/components/rich-text/rich-text.tsx

```diff
@@ -0,0 +1,48 @@
+import { React, renderToString, useStyleSheet } from "../../deps.ts";
+import { Emote } from "../../shared/mod.ts";
+import EmoteRenderer from "../emotes/emote.tsx";
+import stylesheet from "./rich-text.scss.js";
+
+const SPLIT_PATTERN = /[\s.,?!]/;
+
+function splitWords(string: string): string[] {
+  const result: string[] = [];
+  let startOfMatch = 0;
+  for (let i = 0; i < string.length - 1; i++) {
+    if (SPLIT_PATTERN.test(string[i]) !== SPLIT_PATTERN.test(string[i + 1])) {
+      result.push(string.substring(startOfMatch, i + 1));
+      startOfMatch = i + 1;
+    }
+  }
+  result.push(string.substring(startOfMatch));
+  return result;
+}
+
+function formatRichText({ text, emoteMap }: { text: string; emoteMap: Map<string, Emote> }) {
+  const words = splitWords(text);
+  let msg = "";
+  for (const word of words) {
+    const emote = emoteMap?.get(word);
+    if (emote) {
+      msg += renderToString(<EmoteRenderer emote={emote} />);
```

Comment:
this is one of the things I was a little iffy on. Looking for a way to create a dynamic string for the message content. If anyone has suggestions on a better way to do this I'm open to other ideas